src/rgw/rgw_rados.cc

```diff
@@ -426,7 +426,7 @@ void RGWZoneGroup::post_process_params()
   for (map<string, RGWZone>::iterator iter = zones.begin(); iter != zones.end(); ++iter) {
     RGWZone& zone = iter->second;
     zone.log_data = log_data;
-    zone.log_meta = (is_master && zone.id == master_zone);
+    zone.log_meta = (is_master && zone.id == master_zone && zone.name != default_zone_name);
```

Comment:
@cbodley ,another possibility is to check that the current period is empty (default zg and zone don't have a period or realm).
I will also update need_to_log_metadata.